src/monitoring_service/service.py

```diff
@@ -186,3 +188,32 @@ def _process_new_blocks(self, last_block: BlockNumber) -> None:
                         transaction_hash=tx_hash,
                         receipt=receipt,
                     )
+
+    def _purge_old_monitor_requests(self) -> None:
+        """ Delete all old MRs for which still no channel exists.
+
+        Also marks all MRs which have a channel as not waiting_for_channel to
+        avoid checking them again, every time.
+        """
+        with self.context.db.conn:
+            self.context.db.conn.execute(
+                """
+                UPDATE monitor_request SET waiting_for_channel = false
+                WHERE waiting_for_channel
+                  AND EXISTS (
+                    SELECT 1
+                    FROM channel
+                    WHERE channel.identifier = monitor_request.channel_identifier
+                      AND channel.token_network_address = monitor_request.token_network_address
+                  )
+            """
+            )
+            before_this_is_old = datetime.utcnow() - timedelta(minutes=15)
```

Comment:
In other places I needed to use `datetime.now(timezone.utc)` to get proper timezone enabled datetimes, should we do that here as well?

Reply:
Both are without timezone, so we should be fine. All the trouble we had was just related to marshmallow adding a time zone, which caused problems for equality checks and signatures, right? I don't remember the time actually being off in any case.

Reply:
That's true, my question was more if we should care to use this consistently or not.